image/store_test.go

```diff
@@ -70,6 +72,7 @@ func TestRestore(t *testing.T) {
 
 	invalidPattern := id1.Encoded()[1:6]
 	_, err = imgStore.Search(invalidPattern)
+	assert.Check(t, errdefs.IsNotFound(err), "unexpected error type %T", err)
```

Comment:
I (not too long ago) found that you can use `is.ErrorType` for this; it accepts either a _type_ or a function to check the type (which can be `errdefs.IsSOME_ERROR`);

```suggestion
	assert.Check(t, is.ErrorType(err, errdefs.IsNotFound))
```

When using that, a failure looks like this (I changed the test to make it fail, but you get the idea);

```
store_test.go:75: assertion failed: error is No such image: 40b4d (*errors.withStack), not errdefs.IsInvalidParameter
```

Before (current implementation in this PR), it would look like:

```
store_test.go:75: assertion failed: expression is false: errdefs.IsInvalidParameter(err): unexpected error type *errors.withStack
```

Reply:
Oh, that's a really good find, thanks!